rts/Lua/LuaUtils.cpp

```diff
@@ -695,6 +697,48 @@ int LuaUtils::ParseFloat4Vector(lua_State* L, int index, vector<float4>& vec)
 	}
 }
 
+
+bool LuaUtils::ParseResourcePack(lua_State* L, int index, SResourcePack& pack)
+{
+	if (!lua_istable(L, index))
+		return false;
+
+	for (auto [i, resource] : std::views::enumerate(pack)) {
+		// Parse by resource index, like `{ 1.23, 45.6, 789 }`
+		lua_rawgeti(L, index, i+1);
+		resource = luaL_optnumber(L, -1, resource);
+		lua_pop(L, 1);
+
+/* TODO: unitsync could also have a resource handler instance, and
+ * read modrules when trying to read stuff that could be defined as
+ * resource packs. Dedi is out of luck though. No idea about AI. */
+#if !defined UNITSYNC && !defined DEDICATED && !defined BUILDING_AI
+
+		/* Parse by name, like `{ energy = 45.6, blood_mana = 789, metal = 1.23 }`.
+		 * This takes precedence over index-based since it's more explicit. */
+		const auto resourceDef = CResourceHandler::GetInstance()->GetResource(i);
+		if (!resourceDef)
+			continue;
+
+		/* TODO. The resource handler calls the resources Metal and Energy,
+		 * uppercase, but essentially everywhere else they're lowercase so
+		 * here I also adjust to the effective standard. Once resources can
+		 * be customized this will stop being a problem and the name can be
+		 * passed raw, without making a lowercase copy. */
+		std::string resourceName = resourceDef->name;
+		std::ranges::transform(resourceName, resourceName.begin(), [](unsigned char c){ return std::tolower(c); });
```

Comment:
`StringToLower` and `StringToLowerInPlace` impose a slightly lower comprehension cost.